viewing_party/party.py

```diff
@@ -0,0 +1,259 @@
+#Wave_One part 1 
+def create_movie (title, genre, rating):
+    
+    if title and genre and rating:
+        movie_dict = {
+        "title": title, 
+        "genre": genre, 
+        "rating": rating }
+        return movie_dict 
+    else:
+        return None
+
+#Wave_One part 2 
+def add_to_watched(user_data, movie): 
+    user_data["watched"].append(movie)
+    return user_data
+
+
+#Wave_One part 3 
+
+def add_to_watchlist(user_data, movie):
+
+    user_data["watchlist"].append(movie)
+    return user_data
+
+
+#Wave_One part 4 
+
+def watch_movie (user_data, title): 
+
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+
+            user_data["watched"].append(movie)
+            user_data["watchlist"].remove(movie)
+    return user_data
+
+
+#Wave_TWO Part 1 
+
+def get_watched_avg_rating(user_data):
+
+    
+    watched_list = user_data["watched"] 
+    sum = 0 
+    average_rating = 0.0
+    if len(watched_list) == 0 :
+        return average_rating
+    for movie in watched_list:
+
+        sum += movie["rating"]
+    
+    average_rating = sum/len(watched_list)
+    return  average_rating
+    
+#Wave_TWO Part 2 
+
+def get_most_watched_genre(user_data):
+    most_populare_genre =[]
+    if len (user_data["watched"]) == 0:
+        return None 
+    for movie_genre in user_data["watched"]:
+        
+            if "genre" in movie_genre:
+                most_populare_genre.append(movie_genre["genre"])
+            most_common = max(most_populare_genre, key = most_populare_genre.count)
+            print (most_common)
+
+    return most_common
+
+            
+#Completed using a dict but wasn't my first choice
+# def get_most_watched_genre(user_data):
+#     most_popular_genre = {}
+#     if len(user_data["watched"]) == 0:
+#         return None
+#     for movie_genre in user_data["watched"]:
+#         if movie_genre ["genre"] in most_popular_genre:
+#             most_popular_genre[movie_genre["genre"]] += 1
+
+#         else:
+#             most_popular_genre[movie_genre["genre"]] = 1
+
+#     max_value_genre=max(most_popular_genre, key= most_popular_genre.get)
+#     return max_value_genre
+
+#     # https://www.programiz.com/python-programming/methods/built-in/max
```

Comment:
if this isn't the way you want to implement it then you can remove this commented code